layerindex/tools/import_otherdistro.py

```diff
@@ -0,0 +1,789 @@
+#!/usr/bin/env python3
+
+# Import other distro information
+#
+# Copyright (C) 2013, 2018 Intel Corporation
+# Author: Paul Eggleton <paul.eggleton@linux.intel.com>
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+
+import sys
+import os
+import argparse
+import logging
+from datetime import datetime
+import re
+import tempfile
+import glob
+import shutil
+import subprocess
+import string
+import shlex
+import codecs
+from distutils.version import LooseVersion
+
+sys.path.insert(0, os.path.realpath(os.path.join(os.path.dirname(__file__), '..')))
+sys.path.insert(0, os.path.realpath(os.path.join(os.path.dirname(__file__), 'lib')))
+
+import utils
+import recipeparse
+
+logger = utils.logger_create('LayerIndexOtherDistro')
+
+
+class DryRunRollbackException(Exception):
+    pass
+
+
+def get_gourl(goipath):
+    # Much more crude than the original implementation
+    gourl = goipath
+    if not '://' in gourl:
+        gourl = 'https://' + gourl
+    return gourl
+
+
+def update_recipe_file(path, recipe, repodir, raiseexceptions=False):
+    from layerindex.models import Patch, Source
+    from django.db import DatabaseError
+
+    # yes, yes, I know this is all crude as hell, but it gets the job done.
+    # At the end of the day we are scraping the spec file, we aren't trying to build it
+
+    pnum_re = re.compile('-p[\s]*([0-9]+)')
+    configure_re = re.compile('[/%]configure\s')
+
+    try:
+        logger.debug('Updating recipe %s' % path)
+        recipe.pn = os.path.splitext(recipe.filename)[0]
+
+        f = None
+        encodings = ['utf8', 'iso-8859-1', 'gb2312', 'windows-1250', 'windows-1251', 'windows-1252']
+        for e in encodings:
+            try:
+                f = codecs.open(path, 'r', encoding=e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *f*. The resource is allocated by call *codecs.open*. The resource is referenced at lines 63, 70, 73, 75, 76, 80. The resource is closed at line 76. There are other execution paths that do not contain closure statements, for example, when an exception is thrown by *path.basename()*, *StreamReaderWriter.seek()*. To prevent this resource leak, close *f* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).